rust/kernel/lib.rs

```diff
@@ -57,6 +57,9 @@ pub mod file;
 pub mod fs;
 pub mod gpio;
 pub mod hwrng;
+#[cfg(any(CONFIG_I2C, doc))]
+#[doc(cfg(CONFIG_I2C))]
+pub mod i2c;
```

Comment:
Types like `i2c_device_id` are only available when `CONFIG_I2C` is enabled, right? I think documentation will fail if `CONFIG_I2C` is disabled because of this. Rustdoc only suppresses errors within function bodies, not in external interfaces.

Reply:
Will have to test. Should I disable doc generation, or just disable the parts that fail ab leave all that works?

Reply:
For the other modules we disable doc generation entirely. Only disabling the failing parts is nicer from an end-user perspective, but I don't know how much that complicates things.

Reply:
I could potentially even create a mod bindings, that has the structs as stub, so everything can be documented.

Reply:
I only got to test it now. It just works, disabling `CONFIG_I2C` and it seems to just work. If I see it correctly, all constants that I use are not guarded by `IS_ENABLED(CONFIG_I2C)` in the `i2c.h` header, therefore bindgen generates rust symbols  for them, which can be used for documentation. Should I guard that in the bindgen helper header, and only include `i2c.h` when CONFIG_I2C is enabled?

Reply:
Could you please share a link of what you have?

Reply:
How do you mean link? This pr contains the current code version.